addons/mail/models/mail_thread.py

```diff
@@ -500,10 +500,12 @@ def _track_prepare(self, fields_iter):
         if not fnames:
             return
         self.env.cr.precommit.add(self._track_finalize)
+        authors = self.env.cr.precommit.data.setdefault(f'mail.tracking.author.{self._name}', {})
         initial_values = self.env.cr.precommit.data.setdefault(f'mail.tracking.{self._name}', {})
         for record in self:
             if not record.id:
                 continue
+            authors[record.id] = self.env.user.partner_id.id
```

Comment:
Maybe store the partner instead of its id ? Always better to deal with records when possible.